Empire/ProductionQueue.cpp

```diff
@@ -129,9 +129,9 @@ namespace {
                           << "  to stockpile: " << new_contributions;
         }
 
-        if (new_contributions > stockpile_limit &&
+        if ((new_contributions + stockpile_transfer) > stockpile_limit &&
             GetGameRules().Get<bool>("RULE_STOCKPILE_IMPORT_LIMITED"))
-        { new_contributions = stockpile_limit; }
+        { new_contributions = stockpile_limit - stockpile_transfer; }
 
         return starting_stockpile + new_contributions - stockpile_used;
```

Comment:
I think it would be slightly cleaner (and would avoid having this function seem to contain inconsistencies) if we had this return line be 
```
return starting_stockpile + new_contributions + stockpile_transfer - stockpile_used;
```
or after my requested variable rename, 
```
return starting_stockpile + new_contributions + project_transfer_to_stockpile - stockpile_used;
```

and simply did away with line 775
```
 m_expected_new_stockpile_amount += project_transfer_to_stockpile;
```

Or am I overlooking some trouble that would cause somewhere?

Reply:
Sounds  good